### DISCUSSION TITLE
Adding Your Project to Documentation

### DISCUSSION THREAD

**Discussion by thestumbler** [modm-io/modm#1105] [General]
I figured out that I can add `../../src` to `doxyfile.cfg` manually, and it seems to do the trick.
Is there a way to specify this in the `project.xml` file?

**Comment by salkinium** [comment]
No, but you could [add it as an option to the `modm:docs` module](https://github.com/modm-io/modm/blob/develop/tools/doc_generator/module.lb#L83).

**Reply by thestumbler** [reply]
Gotcha. I have still been avoiding modifying the MODM scripts, partly out of fear of breaking something and partly because until recently I was rebuilding the local MODM many times a day. Seems like it's time to take the plunge. 

Is there an easy way to tell Doxygen to generate such project docs into a separate tab at the top level of the HTML file? I don't think making a completely separate docs folder is the right approach, but I'd like to have a way that someone could examine the project docs from a top level page, but still being able to drill down into MODM stuff if needed. Seems like I need to define a new "something" to allow Doxygen to organize them into a separate space. 

Side topic, do you have Doxypress running on your Mac? I can run both flavors on Windows and Ubuntu machines, but I'm only able to get Doxygen working on the Mac.

**Reply by salkinium** [reply]
> Is there an easy way to tell Doxygen to generate such project docs into a separate tab at the top level of the HTML file? 

I'm not sure, but you can certainly put the application stuff into it's own `defgroup`. The modm stuff should all be contained into the modm namespace (if it's not, that's a bug).
I think it's the reason why we intended to let someone use a separate doxygen config for their application with different settings and somehow (?) link to the modm stuff. I honestly haven't gotten there yet.

> Side topic, do you have Doxypress running on your Mac?

Yes, does Homebrew not work for you?

```
brew tap modm-ext/modm
brew install doxypress
```

The [releases are available also for manual install here](https://github.com/copperspice/doxypress/releases).

**Reply by salkinium** [reply]
> partly out of fear of breaking something

Meh, just open a PR if you're stuck and I can directly help you. You'll also get free validation from our CI.

**Reply by thestumbler** [reply]
> Meh, just open a PR if you're stuck and I can directly help you. You'll also get free validation from our CI.

Oh. The CI runs even on my local fork? I'm not well-versed in CI, but that's cool. Do I need to flip any switches or make it happen?

**Reply by thestumbler** [reply]
I was not installing from your tap. But after trying that, I get the same problem. Searching online doesn't show many results, but regardless, it isn't urgent. I can still run DOXYGEN, and I can run DOXYPRESS on other architectures. It is curious.

```
rclott@herman docs$ doxypress doxypress.json

DoxyPress Version: 1.5.1

**  Read Project Configuration
**  Verify Project Configuration


Initialization
Parse input files
Segmentation fault: 11

```
